packages/babel-helpers/src/helpers/applyDecs2311.ts

```diff
@@ -562,16 +580,16 @@ export default /* @no-mangle */ function applyDecs2311(
       return (
         hasClassDecs && [
           defineMetadata(
-            applyDec(
+            (targetClass = applyDec(
```

Comment:
This change fixes the incorrect `this` value previously we passed to the class initializers.